### PR TITLE
fix: [CDS-44534]: dashboard stats optimize

### PR DESCRIPTION
**Pull request by abhinav-harness**
## Describe your changes
The idea behind this is we will create dashboard stat via aggregation. Since it is a massive change we will log in case we see any inconsistency in bg thread in calculation of the values.
If we dont see any inconsistency we will switch on the feature flag and move to newer apis.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38618)

### CODE REVIEW

**Review by Y0709 — APPROVED** [review]
> Reviewed 3 of 5 files at r1, 2 of 3 files at r2, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38618)* status: :shipit: complete! ✓

**Comment by abhinav-harness** [comment]
trigger messagecheck

File: 400-rest/src/main/java/software/wings/resources/StatisticsResource.java [hunk 1]
```diff
@@ -51,6 +78,25 @@ public RestResponse<DeploymentStatistics> deploymentStats(@QueryParam("accountId
...
+      String accountId, Integer numOfDays, List<String> appIds, ServiceInstanceStatistics finalDeploymentStatistics) {
+    ServiceInstanceStatistics serviceInstanceStatistics =
+        statisticsService.getServiceInstanceStatisticsNew(accountId, appIds, numOfDays);
+    if (finalDeploymentStatistics != null && !finalDeploymentStatistics.equals(serviceInstanceStatistics)) {
+      log.error("DEBUG LOG: old way service-instance stats: [{}], new way service-instance stats:[{}]",
+          finalDeploymentStatistics, serviceInstanceStatistics);
+    }
   }
-}
+}
```
Comment (lasdox) [inline 400-rest/src/main/java/software/wings/resources/StatisticsResource.java:102]: new line

Comment (abhinav-harness) [reply 400-rest/src/main/java/software/wings/resources/StatisticsResource.java:102]: its on clang format

**Review by lasdox — APPROVED** [review]
> Reviewed 3 of 5 files at r1, 2 of 3 files at r2, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38618)* status: 1. unresolved discussions.

**Review by lasdox — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38618)* status: :shipit: complete! ✓

**Pull request merged**